### PR TITLE
JAMES-2723 POJOify Mailbox

### PR DESCRIPTION
**Pull request by chibenwa**
- JPAMailbox should act as a DTO representing a mailbox
 - Drop useless Mailbox implementation
 - Merge SimpleMailbox into Mailbox
 - This merge enables some simple refactorings

### CODE REVIEW

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/model/JPAMailbox.java [hunk 1]
```diff
@@ -61,10 +59,18 @@
...
-public class JPAMailbox implements Mailbox {
+public class JPAMailbox {
     
     private static final String TAB = " ";
 
+    public static JPAMailbox from(Mailbox mailbox) {
+        JPAMailbox jpaMailbox = new JPAMailbox(mailbox);
+        /*if (mailbox.getMailboxId() != null) {
+            jpaMailbox.setMailboxId(mailbox.getMailboxId());
+        }*/
```
Comment (rouazana) [inline mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/model/JPAMailbox.java]: to remove?

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSearchIndexTest.java [hunk 2]
```diff
@@ -35,9 +35,7 @@
 import org.apache.james.mailbox.MailboxSession;
 import org.apache.james.mailbox.MailboxSessionUtil;
 import org.apache.james.mailbox.MessageUid;
-import org.apache.james.mailbox.model.Mailbox;
-import org.apache.james.mailbox.model.MailboxACL;
-import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.SimpleMailbox;
```
Comment (rouazana) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMailboxMessageSearchIndexTest.java]: missing a verb in commit comment a18a600399553650e9b0261403da04b319d9157d

**Comment by chibenwa** [comment]
`[INFO] Apache James MPT Imap Mailbox - JPA ................ FAILURE [ 49.362 s]`

Oups I might have broken mailbox rename...

**Comment by chibenwa** [comment]
```
[a747c7baa9f8ef7868c071632b8bd0f4af410880] [ERROR] Errors: 
[a747c7baa9f8ef7868c071632b8bd0f4af410880] [ERROR] org.apache.james.mailbox.MailboxManagerTest$BasicFeaturesTests.renameMailboxShouldChangeTheMailboxPathOfAMailbox
[a747c7baa9f8ef7868c071632b8bd0f4af410880] [ERROR]   Run 1: MailboxManagerTest$BasicFeaturesTests.renameMailboxShouldChangeTheMailboxPathOfAMailbox:1206 » MailboxNotFound
[a747c7baa9f8ef7868c071632b8bd0f4af410880] [ERROR]   Run 2: MailboxManagerTest$BasicFeaturesTests.renameMailboxShouldChangeTheMailboxPathOfAMailbox:1206 » MailboxNotFound
```

I can not wait to deprecate and remove maildir!!!!

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java [hunk 3]
```diff
@@ -19,78 +19,158 @@
...
+import com.google.common.base.Objects;
 
 /**
  * Models long term mailbox data.
  */
-public interface Mailbox {
-    void setMailboxId(MailboxId id);
-    
+public class Mailbox {
+    private MailboxId id = null;
```
Comment (trantienduchn) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java]: should it be Optional?

Comment (chibenwa) [reply mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java]: Nope, I did not intend to refactor this class

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**